### PR TITLE
test: e2e test chash upstream with key (remote_addr) 

### PR DESCRIPTION
**Pull request by idbeta**
Please answer these questions before submitting a pull request

- Why submit this pull request?
- [x] E2E test
- [ ] New feature provided
- [ ] Improve performance

- Related issues
related issue: #908

### CODE REVIEW

File: api/test/e2e/upstream_test.go [hunk 1]
```diff
@@ -188,7 +188,60 @@ func TestRoute_Node_Host(t *testing.T) {
...
+		},
+		{
+			caseDesc:     "hit the route just created",
+			Object:       APISIXExpect(t),
+			Method:       http.MethodGet,
+			Path:         "/server_port",
+			ExpectStatus: http.StatusOK,
+			ExpectBody:   "1981",
+			Sleep:        sleepTime,
+		},
```
Comment (membphis) [inline api/test/e2e/upstream_test.go:237]: delete upstream object

**Review by membphis — CHANGES_REQUESTED** [review]
more test cases:

route contains upstream 
service bind upstream object 
service contains upstream

File: api/test/e2e/upstream_test.go (hunk 1 above)
Comment (idbeta) [reply api/test/e2e/upstream_test.go:237]: so far, delete the upstream operation is completed in the last test case `TestUpstream_Delete`

File: api/test/e2e/upstream_test.go [hunk 2]
```diff
@@ -188,10 +188,63 @@ func TestRoute_Node_Host(t *testing.T) {
...
+                    "port": 1980,
+                    "weight": 1
+				},
+				{
+                    "host": "172.16.238.20",
+                    "port": 1981,
+                    "weight": 2
+                }],
+				"type": "chash",
+				"hash_on":"header",
```
Comment (nic-chen) [inline api/test/e2e/upstream_test.go:210]: need cases for different hash_on

Comment (idbeta) [reply api/test/e2e/upstream_test.go:218]: ok, you can check #932 #936

File: api/test/e2e/upstream_test.go [hunk 3]
```diff
@@ -188,10 +191,198 @@ func TestRoute_Node_Host(t *testing.T) {
...
+		respBody, err = ioutil.ReadAll(resp.Body)
+		body := string(respBody)
+		if _, ok := res[body]; !ok {
+			res[body] = 1
+		} else {
+			res[body] += 1
+		}
+		resp.Body.Close()
+	}
+	assert.Equal(t, 18, res["1980"]+res["1981"]+res["1982"])
```
Comment (membphis) [inline api/test/e2e/upstream_test.go:263]: this way is meanless. 

I think we can write more comments here or we change to a different way.

**Review by membphis — CHANGES_REQUESTED** [review]

**Review by membphis — APPROVED** [review]

**Comment by membphis** [comment]
@nic-chen @moonming @gxthrj @ShiningRush please review this PR if you have time

**Comment by nic-chen** [comment]
LGTM Except for some code format issues.

File: api/test/e2e/upstream_test.go [hunk 4]
```diff
@@ -188,10 +191,198 @@ func TestRoute_Node_Host(t *testing.T) {
...
+		assert.Nil(t, err)
+		respBody, err = ioutil.ReadAll(resp.Body)
+		body := string(respBody)
+		if _, ok := res[body]; !ok {
+			res[body] = 1
+		} else {
+			res[body] += 1
+		}
+		resp.Body.Close()
+	}
```
Comment (nic-chen) [inline api/test/e2e/upstream_test.go:262]: Please add todo to remember to improve the code format, thanks.

**Review by nic-chen — APPROVED** [review]

File: api/test/e2e/upstream_test.go (hunk 3 above)
Comment (idbeta) [reply api/test/e2e/upstream_test.go]: fixed.

File: api/test/e2e/upstream_test.go (hunk 4 above)
Comment (idbeta) [reply api/test/e2e/upstream_test.go:263]: fixed.

**Pull request merged**